lib/web_ui/lib/src/engine/text/line_breaker.dart

```diff
@@ -43,62 +28,117 @@ class LineBreakResult {
   final LineBreakType type;
 }
 
+/// Normalizes properties that behave the same way into one common property.
+LineCharProperty _normalizeLineProperty(LineCharProperty prop) {
```

Comment:
Can this substitution be made in the `_packedLineBreakProperties` so we never need to convert at run time?

Reply:
This normalization is part of the algorithm. I want to keep codegen independent.

The other reason is that in some cases we still need to know the original property (pre-normalization). e.g. `rawCurr == LineCharProperty.LF`.

There are a few optimizations that I can think of. I expect them to have marginal perf improvements but nothing drastic.

Some ideas for my future reference:
- Normalize properties during codegen.
- Reduce the amount of binary searches by either caching at runtime, or codegen'ing a hash map for the most common characters, or both.
- For paragraphs that can fit in a single line, we can avoid a whole bunch of computations.

Reply:
How about inside `UnicodePropertyLookup.fromPackedData`?

I'm just throwing ideas out. The PR LGTM. This comment is non-blocking.

Reply:
`UnicodePropertyLookup.fromPackedData` is used by both line breaker and word breaker. They have different properties and different rules for normalization.

Reply:
I see your general point though. And I agree we could do some optimizations there.